src/Cake.Common/Tools/NuGet/List/NuGetList.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Cake.Core;
+using Cake.Core.IO;
+using Cake.Core.IO.NuGet;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.NuGet.List
+{
+    /// <summary>
+    /// The NuGet package lister used to list NuGet packages from a source.
+    /// </summary>
+    public sealed class NuGetList : NuGetTool<NuGetListSettings>
+    {
+        private readonly ICakeEnvironment _environment;
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="NuGetList"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="tools">The tool locator.</param>
+        /// <param name="resolver">The NuGet tool resolver.</param>
+        public NuGetList(
+            IFileSystem fileSystem,
+            ICakeEnvironment environment,
+            IProcessRunner processRunner,
+            IToolLocator tools,
+            INuGetToolResolver resolver) : base(fileSystem, environment, processRunner, tools, resolver)
+        {
+            _environment = environment;
+        }
+
+        /// <summary>
+        /// Lists available packages with their versions
+        /// </summary>
+        /// <param name="packageId">The source package id.</param>
+        /// <param name="settings">The settings</param>
+        /// <returns>A list of available packages</returns>
+        public IEnumerable<NuGetListItem> List(string packageId, NuGetListSettings settings)
+        {
+            if (packageId == null)
```

Comment:
This check is redundant. Next check will also catch `null`

Reply:
Removed the second check instead (see implementation of omiting the packageId)